src/packages/core/icon-registry/icon.registry.ts

```diff
@@ -13,13 +12,21 @@ interface UmbIconDescriptor {
  * @description - Icon Registry. Provides icons from the icon manifest. Icons are loaded on demand. All icons are prefixed with 'icon-'
  */
 export class UmbIconRegistry extends UUIIconRegistry {
+	#icons: UmbIconDescriptor[] = [];
+
+	setIcons(icons: UmbIconDescriptor[]) {
+		this.#icons = icons;
+	}
+	appendIcons(icons: UmbIconDescriptor[]) {
+		this.#icons = [...this.#icons, ...icons];
+	}
 	/**
 	 * @param {string} iconName
 	 * @return {*}  {boolean}
 	 * @memberof UmbIconStore
 	 */
 	acceptIcon(iconName: string): boolean {
-		const iconManifest = icons.find((i: UmbIconDescriptor) => i.name === iconName);
+		const iconManifest = this.#icons.find((i: UmbIconDescriptor) => i.name === iconName);
```

Comment:
I think because this isn't an async method, but the icons are loaded asynchronously, you will end up with cases where an icon is loaded after it has been requested and so will never show up. There seems to be no check in this registry to check whether the observer in UmbIconRegistryContext has been run or not, or is there?

Reply:
Oh, yes this is a problem. I will see what our options are.

Reply:
There is a fix now.